numpy/core/_internal.py

```diff
@@ -263,6 +273,16 @@ def __init__(self, array, ptr=None):
         else:
             self._zerod = False
 
+    def __getattr__(self, name):
+        # Numpy 1.21.0, 2021-05-18
+        if name in _CTYPES_DEPRECATED:
+            name_new = _CTYPES_DEPRECATED[name]
+            warnings.warn(f"{name!r} is deprecated. Use {name_new!r} instead",
+                          DeprecationWarning, stacklevel=2)
+            return getattr(type(self), name_new).fget
+        else:
+            return super().__getattribute__(name)
```

Comment:
I think it might end up as fewer lines to just write out each function individually, which also leaves a place for a docstring with `..deprecated`

Reply:
It's a bit longer (https://github.com/numpy/numpy/pull/19031/commits/a1056030804b0f204d35f579e4da4585ac3c150e), but also a lot more explicit now.